### PR TITLE
Fix stack overflow exception in deleteOutput

### PR DESCRIPTION
**Pull request by Warchant**
Fixes https://github.com/bazelbuild/bazel/issues/14393

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/actions/AbstractAction.java [hunk 1]
```diff
@@ -497,6 +497,12 @@ public static void deleteOutput(Path path, @Nullable Root root) throws IOExcepti
         // Retry deleting after making the parent writable.
         parentDir.setWritable(true);
         deleteOutput(path, root);
```
Comment (meteorcloudy) [inline src/main/java/com/google/devtools/build/lib/actions/AbstractAction.java:499]: This recursion call here is really confusing and unneccessary. I think we can just replace it with:
```
if (path.isDirectory(Symlinks.NOFOLLOW)) {
  path.deleteTree();
} else {
  path.delete();
}
```

Then we don't need the special check here.

**Review by meteorcloudy — CHANGES_REQUESTED** [review]
Thanks!

File: src/main/java/com/google/devtools/build/lib/actions/AbstractAction.java (hunk 1 above)
Comment (Warchant) [reply src/main/java/com/google/devtools/build/lib/actions/AbstractAction.java]: Done

**Comment by Warchant** [comment]
Is PR merged or just closed?

**Comment by meteorcloudy** [comment]
Merged at https://github.com/bazelbuild/bazel/commit/1746d6341d22278a615a3707f2ab9e5a9d83df0b!

**Comment by meteorcloudy** [comment]
We have to import this change to internal code base, then export to github.

**Pull request closed without merging**